crypto/err/err.c

```diff
@@ -812,3 +812,26 @@ int ERR_pop_to_mark(void)
     es->err_flags[es->top] &= ~ERR_FLAG_MARK;
     return 1;
 }
+
+int ERR_clear_last_mark(void)
+{
+    ERR_STATE *es;
+    int top;
+
+    es = ERR_get_state();
+    if (es == NULL)
+        return 0;
+
+    top = es->top;
+    while (es->bottom != top
+           && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0) {
+        top -= 1;
+        if (top == -1)
+            top = ERR_NUM_ERRORS - 1;
```

Comment:
As for ERR_set_error_data, it makes even less sense... Who is advancing `es->top`? Zero-slot is never used? But see rather ERR_put_error for "perfect" definition of circular buffer... Too bad it can't be used when you traverse backwards...